lib/puppet/util/log.rb

```diff
@@ -284,47 +292,34 @@ def to_hash
   end
 
   def to_data_hash
-    {
+    hash = {
       'level' => @level,
       'message' => @message,
       'source' => @source,
       'tags' => @tags,
       'time' => @time.iso8601(9),
-      'file' => @file,
-      'line' => @line,
     }
+    [:backtrace, :environment, :node, :issue_code, :file, :line, :pos].each do |attr|
+      iv = "@#{attr}"
+      hash[attr.to_s] = instance_variable_get(iv) if instance_variable_defined?(iv)
+    end
+    hash
   end
 
   def to_pson(*args)
     to_data_hash.to_pson(*args)
   end
 
-  def message=(msg)
-    raise ArgumentError, "Puppet::Util::Log requires a message" unless msg
-    @message = msg.to_s
-  end
-
-  def level=(level)
```

Comment:
On the other hand, there are not calls that does that. Not in our code anyway and that's where the filtering on level is performed. Do we want to support possible third monkey patching?